Framework/Algorithms/src/ReflectometryReductionOne.cpp

```diff
@@ -578,29 +590,30 @@ void ReflectometryReductionOne::exec() {
     auto detectorWS = inLam.get<0>();
     auto monitorWS = inLam.get<1>();
 
-    if (isMultiDetector) {
-      if (directBeam.is_initialized()) {
-        // Sum over the direct beam.
-        WorkspaceIndexList db = directBeam.get();
-        std::stringstream buffer;
-        buffer << db.front() << "-" << db.back();
-        MatrixWorkspace_sptr regionOfDirectBeamWS = this->toLamDetector(
-            buffer.str(), runWS, wavelengthInterval, wavelengthStep);
+    if (detectorEfficiencyCorrection.is_initialized()) {
+      MatrixWorkspace_sptr decWS = *detectorEfficiencyCorrection;
+      if (decWS->blocksize() > 1) {
+        // If using separate normalization constants for each spectra, rebin
+        // to the detector workspace
+        if (decWS->blocksize() < detectorWS->blocksize()) {
+          throw std::invalid_argument("The number of y-values in the detector "
```

Comment:
The issue at heart is that in the original code, both the `RegionOfDirectBeam` WS (now `decWS`) and the `detector` WS were both provided as 1-spectra many values workspaces. The blocksize checks were to ensure the `decWS` wasn't smaller than the `detectorWS` it was being applied to. However, `decWS` is now provided externally and is in many-spectra 1-value format. I will deal with this by attempting to convert it into the original form - 1-spectra many-values.